### ISSUE TITLE
Android build - Cannot fit requested classes in a single dex file

### ISSUE THREAD

**Issue by LcsLpr** [tinycreative/react-native-intercom#199]
Hello,

I'am using "react-native-intercom": "^10.2.0" but I cannot build with it ...
When I try to build, I have a strange error :

> > Task :app:transformDexArchiveWithExternalLibsDexMergerForDebug FAILED
> D8: Cannot fit requested classes in a single dex file. Try supplying a main-dex list.
> methods: 68334 > 65536

Adding : "multiDexEnabled true" in build.gradle permit to build, but the app crash directly with some others dependencies (like OneSignal) ... So I don't think it is a good solution to fix this ...

This is my dependencies :
```
dependencies {
    implementation project(':react-native-intercom')
    implementation project(':react-native-linear-gradient')
    implementation project(':react-native-i18n')
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.facebook.react:react-native:+"
    implementation project(':tipsi-stripe')
    implementation (project(':react-native-maps')) {
        exclude group: 'com.google.android.gms', module: 'play-services-base'
        exclude group: 'com.google.android.gms', module: 'play-services-maps'
    }
    implementation (project(':react-native-camera')) {
        exclude group: "com.google.android.gms"
        exclude group: "com.android.support", module: 'support-v4'
    }
    implementation "com.android.support:appcompat-v7:23.0.1"
    implementation 'com.google.android.gms:play-services-base:10.0.1'
    implementation 'com.google.android.gms:play-services-maps:10.0.1'
}
```

**Comment by pierregoutheraud** [comment]
@LucasLpr Did you find any solution ? 🙏

**Comment by ecexplorer** [comment]
Other context for people that experience this like I did, react native starts with about 30,000 dex references. Intercom adds another 20,000, which puts apps pretty close to the 65,000 limit. This has nothing to do with react-native-intercom, but with intercom's sdk itself. I've opened a topic on their discussion board asking for alternatives https://community.intercom.com/t/20000-dex-references/765
